### PR TITLE
Improve general code quality

### PR DESCRIPTION
**Pull request by yanhwee**

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AddGoodsCommandParser.java [hunk 1]
```diff
@@ -36,13 +36,21 @@ public AddGoodsCommand parse(String args) throws ParseException {
                 ArgumentTokenizer.tokenize(args, PREFIX_GOODS_NAME, PREFIX_QUANTITY, PREFIX_PRICE,
                         PREFIX_CATEGORY, PREFIX_PROCUREMENT_DATE, PREFIX_ARRIVAL_DATE, PREFIX_NAME);
 
-        if (!arePrefixesPresent(argMultimap, PREFIX_GOODS_NAME, PREFIX_QUANTITY, PREFIX_PRICE, PREFIX_CATEGORY,
-                PREFIX_PROCUREMENT_DATE, PREFIX_ARRIVAL_DATE, PREFIX_NAME) || !argMultimap.getPreamble().isEmpty()) {
+        if (!arePrefixesPresent(argMultimap, PREFIX_GOODS_NAME, PREFIX_QUANTITY, PREFIX_PRICE,
+                PREFIX_CATEGORY, PREFIX_PROCUREMENT_DATE, PREFIX_ARRIVAL_DATE, PREFIX_NAME)
+                || !argMultimap.getPreamble().isEmpty()) {
```
Comment (jialegitgud) [inline src/main/java/seedu/address/logic/parser/AddGoodsCommandParser.java:41]: Should be expecting another indent here

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java [hunk 2]
```diff
@@ -34,23 +34,35 @@ public EditCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
                 ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_ADDRESS, PREFIX_TAG);
 
+        Name targetName = parseName(argMultimap);
+
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_ADDRESS);
```
Comment (jialegitgud) [inline src/main/java/seedu/address/logic/parser/EditCommandParser.java:39]: Was this check previously implemented?

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 3]
```diff
@@ -26,25 +26,34 @@ public FindCommand parse(String args) throws ParseException {
...
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
-        }
+    private static List<String> parseKeywords(ArgumentMultimap argMultimap) {
+        String keywordsArg = argMultimap.getPreamble().trim();
 
-        return new FindCommand(keywords, categorySet);
+        return Arrays.stream(keywordsArg.split("\\s+"))
+                .filter(x -> !x.isEmpty())
+                .collect(Collectors.toList());
```
Comment (jialegitgud) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java:57]: Might want to use `streams.toList()` directly here.

File: src/main/java/seedu/address/logic/parser/ListGoodsCommandParser.java [hunk 4]
```diff
@@ -37,39 +35,51 @@ public boolean test(GoodsReceipt gr) {
...
-        // default predicate to return all items
-        if (predicate == null) {
-            predicate = DUMMY_PREDICATE;
-        }
+        Predicate<GoodsReceipt> predicate = parsePredicate(argMultimap);
 
         return new ListGoodsCommand(predicate);
     }
 
+    private static Predicate<GoodsReceipt> parsePredicate(ArgumentMultimap argMultimap) throws ParseException {
```
Comment (jialegitgud) [inline src/main/java/seedu/address/logic/parser/ListGoodsCommandParser.java:51]: Good use of streams to rewrite this messy implementation! 👍

File: src/main/java/seedu/address/model/goods/Goods.java [hunk 5]
```diff
@@ -3,45 +3,21 @@
 import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
 
 /**
- * Represents goods.
- * Garauntees: Immutable; name is valid as declared in {@link #isValidGoodsName(String)};
+ * An immutable class that represents goods.
  */
-public class Goods {
+public record Goods(GoodsName goodsName, GoodsCategories category) {
```
Comment (jialegitgud) [inline src/main/java/seedu/address/model/goods/Goods.java:8]: Might not be necessary, as we don't want to risk additional bugs using the `Record` identifier.

**Review by jialegitgud — APPROVED** [review]
Overall LGTM 💯 , just make sure where some code quality changes are introduced the relevant checks have been manually done to make sure it doesn't change the features in an unintended manner.

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java (hunk 2 above)
Comment (yanhwee) [reply src/main/java/seedu/address/logic/parser/EditCommandParser.java:39]: Nope. This is kind of a minor bug fix.

**Pull request merged**